From d828b04de880f69f03961e5b2c4a19aea0b69c49 Mon Sep 17 00:00:00 2001
From: Myy <myy@miouyouyou.fr>
Date: Tue, 10 Jan 2017 03:51:40 +0000
Subject: [PATCH] mmc: Applied Ziyuan Xu dw_mmc patch

This fixes the boot-time hang-up.
See here for details about the patch:
https://patchwork.kernel.org/patch/9498527/

See here for details about the problem solved:
https://bugzilla.kernel.org/show_bug.cgi?id=192131

Signed-off-by: Myy <myy@miouyouyou.fr>
---
 drivers/mmc/host/dw_mmc.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/mmc/host/dw_mmc.c b/drivers/mmc/host/dw_mmc.c
index b44306b..192e586 100644
--- a/drivers/mmc/host/dw_mmc.c
+++ b/drivers/mmc/host/dw_mmc.c
@@ -3354,10 +3354,12 @@ int dw_mci_runtime_resume(struct device *dev)
 
 		if (!slot)
 			continue;
-		if (slot->mmc->pm_flags & MMC_PM_KEEP_POWER) {
+		if (slot->mmc->pm_flags & MMC_PM_KEEP_POWER)
 			dw_mci_set_ios(slot->mmc, &slot->mmc->ios);
-			dw_mci_setup_bus(slot, true);
-		}
+		/* Force setup bus to guarantee available clock output 
+*/
+		dw_mci_setup_bus(slot, true);
+
 	}
 
 	/* Now that slots are all setup, we can enable card detect */
-- 
2.10.2

